src/spaniel-observer.ts

```diff
@@ -179,19 +193,20 @@ export class SpanielObserver implements SpanielObserverInterface {
     });
   }
   private handleThresholdExiting(spanielEntry: SpanielObserverEntry, state: SpanielThresholdState) {
-    let { time } = spanielEntry;
+    let { highResTime } = spanielEntry;
     let hasTimeThreshold = !!state.threshold.time;
     if (state.lastSatisfied && (!hasTimeThreshold || (hasTimeThreshold && state.visible))) {
       // Make into function
-      spanielEntry.duration = time - state.lastVisible;
+      spanielEntry.duration = highResTime - state.lastVisible.highResTime;
+      spanielEntry.visibleTime = state.lastVisible.unixTime;
       spanielEntry.entering = false;
       state.visible = false;
       this.queuedEntries.push(spanielEntry);
     }
 
     clearTimeout(state.timeoutId);
   }
-  private handleObserverEntry(entry: IntersectionObserverEntry) {
+  private handleObserverEntry(entry: InternalIntersectionObserverEntry) {
```

Comment:
I'm not sure about this.
`handleObserverEntry` is bound as IO callback, and receives hrTime, why switch to use `InternalIntersectionObserverEntry` which has extra fields? And it seems we generates unix time inside spaniel, but `internalCallback` will pass in hrTime.